src/views/Generator/Generator.utils.ts

```diff
@@ -47,9 +47,30 @@ export const writeGeneratorToFile = (
   )
 }
 
-export const loadGeneratorFile = async (fileName: string) => {
+export const isGeneratorDirty = (
+  generatorStateData: Partial<GeneratorFileData>,
+  generatorFileData: Partial<GeneratorFileData>
+) => {
+  // Convert to JSON instead of doing deep equal to remove
```

Comment:
I have a question not related to this particular PR:
I think it shouldn't be a problem here, but since JSON.stringify does not guarantee the order of keys, can two equal objects be considered unequal?

Reply:
That's possible. We might want to switch to `isEqual` if this proves to be a problem.

Reply:
IIRC, I've used JSON.stringify because generatorFileData would include all the undefined values, but stored JSON wouldn't have them, and if would show up as false dirty. So we'd need to do some cleanup if we switch to `isEqual`